GVRf/Framework/framework/src/main/java/org/gearvrf/GVRSceneObject.java

```diff
@@ -270,7 +270,13 @@ public GVRSceneObject(GVRContext gvrContext, float width, float height,
 
     protected void finalize() throws Throwable
     {
-        detachAllComponents();
+        try {
+            detachAllComponents();
+        } catch(final Exception exc) {
+            exc.printStackTrace();
```

Comment:
Any RuntimeExceptions. A finalizer should not throw exceptions and and should also ensure the super.finalize is invoked. I have seen this finalizer throw occasionally.